packages/core/lib/commands/debug/loadConfig.js

```diff
@@ -0,0 +1,24 @@
+module.exports = function (options) {
+  const Config = require("@truffle/config");
+  const TruffleError = require("@truffle/error");
+
+  let config;
+  try {
+    config = Config.detect(options);
+  } catch (error) {
+    if (error instanceof TruffleError && options.url) {
+      // in case config file is not detected (exception thrown) AND url is provided in the options,
+      // We use default config and set compileNone to true. Since there are is no config files and url is provided,
+      // It is assumed that truffle debug is being used for analysis and debugging and that there is nothing to compile.
+      // E.g. analysing/debugging a single transaction of an external project
+      config = Config.default();
+      config.compileNone = true;
+      // forcing options to also set compileNone to true, otherwise it can be later override on Config.merge(...)
+      options.compileNone = true;
```

Comment:
Oh, hm, you fixed it by mutating the `options` that was passed in... I don't think this is a good idea, it means that `loadConfig` has a hidden side effect.  (I know, some of these functions have the hidden side effect of altering `config`, but that we just have to accept.)  Can you fix it without mutating the `options` that was passed in?  A quick way if you want to keep the code this way would be to clone and have this return `{ config, options }` instead of just `config`, maybe not the best way, but just throwing things out...

Reply:
actually, I had done some changes meanwhile.
I agree it doesn't look good.
Also, I am not happing returning `config` and `options`

So I did a couple of changes
1. Remove the `mergeConfigNetwork` from `run.js`. Is there a use-case we want to manually add a provider but we can't set `--network` as an argument?
2. Added `mergeConfigNetwork` here to `loadConfig`. So we merge and `options` to defaul config and then set `compileNone` to `true`

wdyt?

Reply:
Ah!  Yes, I agree that is a better structure.  However there's no `mergeConfigNetwork` in that first branch, there should be one there too, right?

Reply:
What do you mean in the first branch? do you mean on the `try` block? 
Why should we have it there? if `url` is provided and we have a config file, we basically ignore `url` anyway, right?

To be honest it might make sense to log a warning, saying the parameter is ignored. But other than that, I don't see a reason. Did I miss something?

Reply:
Yes, I mean in the `try` block.  My understanding that the desired behavior was that `url` is not ignored in such case, just the network is overridden.

Reply:
I see what you mean. Updating...

Reply:
should be fine now. Please, let me know of any further point